### PR TITLE
fix pod status ready condition

### LINKED ISSUE #92378: Pod status ready condition is incorrect
**What happened**:
- kubelet pod status cache ready condition is `true` and node system hangs
- controller-manager update pod status ready condition to `false` because node `NotReady`
- node system recovery and kubelet update node status to `Ready`
- but kubelet doesn't update pod status ready condition to `true`, because cache is `true`

**What you expected to happen**:
kubelet update pod status ready condition to `true` after node recovery

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by liuxu623**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
We should update pod status when cache is expired 

**Which issue(s) this PR fixes**:

Fixes #92378

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by tedyu** [comment]
Can you clarify the problem this PR solves (via unit / e2e test) ?

thanks

**Comment by liuxu623** [comment]
> Can you clarify the problem this PR solves (via unit / e2e test) ?
> 
> thanks

I have add some test, thank you for help me review code.

File: pkg/kubelet/status/status_manager.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 	normalizeStatus(pod, &status)
 	// The intent here is to prevent concurrent updates to a pod's status from
 	// clobbering each other so the phase of a pod progresses monotonically.
-	if isCached && isPodStatusByKubeletEqual(&cachedStatus.status, &status) && !forceUpdate {
+	if isCached && isPodStatusByKubeletEqual(&pod.Status, &status) && isPodStatusByKubeletEqual(&cachedStatus.status, &status) && !forceUpdate {
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/status/status_manager.go:434]: is it ok that `updateLastTransitionTime ` will not update the transition time in this case?

**Comment by SergeyKanzhelev** [comment]
@liuxu623 this seems to be the right direction to fix the bug. Would you please take a look at the comment - the change may be a bit bigger to also include updating the last transition time variable.

**Comment by liuxu623** [comment]
@SergeyKanzhelev Hi, I have update commit, please help me review, thank you.

**Pull request closed without merging**